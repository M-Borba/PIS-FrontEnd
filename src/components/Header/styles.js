import { makeStyles } from "@material-ui/core/styles";
import styled from "@emotion/styled";

export const Navbar = styled.div`
  height: 74px;
  padding: 0px 40px;
  background-color: #252525;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const UserName = styled.div`
  height: 25px;
  width: fit-content;
  background-color: #fff;
  font-family: Nunito Sans;
  font-style: normal;
  font-weight: 600;
  font-size: 15px;
  line-height: 27px;
  color: #252525;
  padding: 5px 20px;
  border-radius: 50px;
  cursor: pointer;
`;

export const Button = styled.div`
  padding: 0px 20px;
  border-radius: 0;
  text-transform: none;
  font-family: Nunito Sans;
  font-weight: 700;
  height: 94%;
  display: flex;
  justify-content: center;
  align-items: center;
  &:hover {
    border-bottom: 4px #6b5ecd solid;
  }
`;

export const useStyles = makeStyles((theme) => ({
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    height: 74,
    display: "flex",
  },
  link: {
    textDecoration: "none",
    color: "white",
  },
  menuLink: {
    textDecoration: "none",
    color: "black",
  },
  mr5: {
    marginRight: 5,
  },
  mt35: {
    marginTop: 35,
  },
}));

export const PaperProps = {
  elevation: 0,
  sx: {
    overflow: "visible",
    filter: "drop-shadow(0px 2px 8px rgba(0,0,0,0.32))",
    mt: 1.5,
    "& .MuiAvatar-root": {
      width: 32,
      height: 32,
      ml: -0.5,
      mr: 1,
    },
    "&:before": {
      content: '""',
      display: "block",
      position: "absolute",
      top: 0,
      right: 14,
      width: 10,
      height: 10,
      bgcolor: "background.paper",
      transform: "translateY(-50%) rotate(45deg)",
      zIndex: 0,
    },
  },
};

export const anchorElPoint = {
  horizontal: "center",
  vertical: "bottom",
};
