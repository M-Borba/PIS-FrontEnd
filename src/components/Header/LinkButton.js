import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

import { useStyles } from "./styles";

export const LinkButton = ({ src, alt, title, to, isSelected }) => {
  const classes = useStyles();
  const selectedStyle = isSelected ? classes.selected : "";

  return (
    <Link className={`${classes.link} ${selectedStyle}`} to={to}>
      <div className={classes.button}>
        <img style={{ marginRight: 9 }} src={src} alt={alt} />
        {title}
      </div>
    </Link>
  );
};

LinkButton.propTypes = {
  src: PropTypes.node.isRequired,
  alt: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  isSelected: PropTypes.bool.isRequired,
};
