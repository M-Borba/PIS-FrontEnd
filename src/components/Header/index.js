import React, { useState, Fragment } from "react";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import AccountCircleIcon from "@mui/icons-material/AccountCircle";
import LogoutIcon from "@mui/icons-material/Logout";
import Divider from "@material-ui/core/Divider";
import MenuItem from "@material-ui/core/MenuItem";
import Menu from "@material-ui/core/Menu";
import { useStyles, PaperProps, anchorElPoint } from "./styles";
import { BrowserRouter as Router, Link } from "react-router-dom";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import { axiosInstance } from "../../config/axios";
import { Tooltip } from "@material-ui/core";

export default function Header() {
  var uid = localStorage.getItem("uid");
  if (uid == null) {
    uid = "Aún no inició sesión";
  }
  const classes = useStyles();
  const [username] = useState(uid);
  const [anchorEl, setAnchorEl] = useState(null);
  const open = Boolean(anchorEl);

  const handleMenu = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  const Logout = () => {
    axiosInstance
      .delete("/users/sign_out")
      .then(() => {
        //logout en la api
        window.location.reload();
      })
      .catch((error) => {
        console.log(error);
      });
    localStorage.removeItem("token"); //borrado local
    localStorage.removeItem("client");
    localStorage.removeItem("uid");
  };

  return (
    <div className={classes.root}>
      <AppBar style={{ background: '#1c1c1c' }} position="static">
        <Toolbar>
          <Typography variant="h6" className={classes.title}>
            <Button color="inherit">
              <Link className={classes.link} to="/inicio">
                Inicio
              </Link>
            </Button>
            <Button color="inherit">
              <Link className={classes.link} to="/proyectos">
                Proyectos
              </Link>
            </Button>
            <Button color="inherit">
              <Link className={classes.link} to="/personas">
                Personas
              </Link>
            </Button>
            <Button color="inherit">
              <Link className={classes.link} to="/administradores">
                Administradores
              </Link>
            </Button>
          </Typography>
          <Fragment>
            <Tooltip title="Configuración de la cuenta">
              <IconButton
                aria-label="account of current user"
                aria-controls="menu-appbar"
                aria-haspopup="true"
                onClick={handleMenu}
                color="inherit"
                data-testid="menu-perfil"
              >
                <AccountCircleIcon className={classes.mr5} />
                {username}
              </IconButton>
            </Tooltip>
            <Menu
              className={classes.mt35}
              anchorEl={anchorEl}
              getContentAnchorEl={null}
              open={open}
              onClose={handleClose}
              onClick={handleClose}
              PaperProps={PaperProps}
              transformOrigin={anchorElPoint}
              anchorOrigin={anchorElPoint}
            >
              <MenuItem>
                <LogoutIcon fontSize="small" className={classes.mr5} />
                <Link
                  className={classes.menuLink}
                  onClick={Logout}
                  to="/login"
                  data-testid="logout"
                >
                  Cerrar Sesión
                </Link>
              </MenuItem>
            </Menu>
          </Fragment>
        </Toolbar>
      </AppBar>
    </div>
  );
}
