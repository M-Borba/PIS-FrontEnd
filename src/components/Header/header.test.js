import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Header from "./index";
import React from "react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

it("renders correctly unloged", () => {
  const history = createMemoryHistory();
  const { container } = render(
    <Router history={history}>
      <Header />
    </Router>
  );
  // Matches the last snapshot taken
  expect(container).toMatchSnapshot();
});

it("renders and redirects when loged", () => {
  // in a real test a renderer like "@testing-library/react"
  // would take care of setting up the DOM elements
  const history = createMemoryHistory();
  localStorage.setItem("token", "fake-token");
  localStorage.setItem("client", "fake-client");
  localStorage.setItem("uid", "fake@uid.com");
  // Render app
  const { container } = render(
    <Router history={history}>
      <Header />
    </Router>
  );
  // mock push function
  history.push = jest.fn();

  // Matches the last snapshot taken
  expect(container).toMatchSnapshot();

  fireEvent.click(screen.getByText("Inicio"));
  expect(history.push).toHaveBeenCalledWith("/inicio");

  fireEvent.click(screen.getByText("Personas"));
  expect(history.push).toHaveBeenCalledWith("/personas");

  fireEvent.click(screen.getByText("Cerrar Sesi√≥n"));

  expect(history.push).toHaveBeenCalledWith("/login");
});
