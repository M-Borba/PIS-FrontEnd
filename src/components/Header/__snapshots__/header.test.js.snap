// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component tests renders correctly 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <a
              class="makeStyles-link-4"
              href="/login"
            >
              Iniciar Sesion
            </a>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </div>
</div>
`;
